File: ios/PreloadInfo.swift

import ExpoModulesCore
import AppMetricaCore

struct PreloadInfo: Record {
    @Field
    var trackingID: String = ""

    @Field
    var additionalParams: Dictionary<String, String>? = nil

    func toPreloadInfo() -> AppMetricaPreloadInfo {
        let preloadInfo = AppMetricaPreloadInfo(trackingIdentifier: trackingID)
        additionalParams?.forEach({ (key: String, value: String) in
            preloadInfo?.setAdditionalInfo(info: key, for: value)
        })
        return preloadInfo!
    }
}

File: android/src/main/java/io/appmetrica/analytics/expo/models/PreloadInfo.kt
package io.appmetrica.analytics.expo.models

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import io.appmetrica.analytics.PreloadInfo

class PreloadInfo: Record {
    @Field
    val trackingID: String = ""

    @Field
    val additionalParams: Map<String,String>? = null

    fun toPreloadInfo(): PreloadInfo {
        val preloadInfo = PreloadInfo.newBuilder(trackingID)
        additionalParams?.forEach {
            preloadInfo.setAdditionalParams(it.key,it.value)
        }
        return preloadInfo.build()
    }
}